include/solvers/navier_stokes_base.h

```diff
@@ -241,16 +241,20 @@ class NavierStokesBase : public PhysicsSolver<VectorType>
         if (!restart)
           {
             multiphysics->set_initial_conditions();
-            this->postprocess_fd(true);
-            multiphysics->postprocess(true);
-            this->write_output_results(this->present_solution);
           }
         ref_iter++;
       }
     while (
       ref_iter <
         (this->simulation_parameters.mesh_adaptation.initial_refinement + 1) &&
       restart == false);
+
+    if (!restart)
+      {
+        this->postprocess_fd(true);
+        multiphysics->postprocess(true);
+        this->write_output_results(this->present_solution);
```

Comment:
Unrelated to this PR, but should we always output the initial condition even when the user does not request it? For instance, when running application tests this would still generate output files of the initial condition even if the `output_frequency` is set to 0, no?
```suggestion
        bool output_requested =
          ((this->simulation_parameters.simulation_control.output_control ==
              Parameters::SimulationControl::OutputControl::iteration &&
            this->simulation_parameters.simulation_control
                .output_iteration_frequency != 0) ||
           (this->simulation_parameters.simulation_control.output_control ==
              Parameters::SimulationControl::OutputControl::time &&
            this->simulation_parameters.simulation_control
                .output_time_frequency != -1));
        if (output_requested)
          this->write_output_results(this->present_solution);
```